docs/Development-Guides/Rate-Limits.md

```diff
@@ -29,3 +22,74 @@ Product | Max requests initiated per minute
 `employment` | 4
 `payment` | 2
 `pay-statement` | 2
+
+If an access token rate limit is encountered, it will contain the `finch_code`: [finch_token_rl](/docs/Development-Guides/Errors/Error-Types.md#error-types-1) in the response body.
+
+```json
+// HTTP 429 response body for access token rate limit exceeded
+{
+    “statusCode”: 429,
+    “status”: 429,
+    “code”: 429,
+    “message”: “Too many requests for token”,
+    “name”: “rate_limit_exceeded_error”,
+    “finch_code”: “finch_token_rl”
+}
+```
+
+## Application Rate Limits
+
+Similarly, multiple access tokens can be created from a single Finch application as more employers are connected. A Finch application has its own rate limits separate from the access token rate limits. You can think of an application like a development environment or application stage. Finch will assign three types of applications: `sandbox`, `development`, and `production`. Each will have their own `client_id` which represents the application.
+
+For application-level rate limits, each `product` manages another “larger” bucket simultaneously counting all requests across all access tokens created by that application. Each bucket is still scoped to a product endpoint like with access tokens; the only difference is that the application-level bucket is larger.
+
+Product | Max requests initiated per minute
+-------|-------------
+`company` | 20
+`directory` | 20
+`individual` | 20
+`employment` | 20
+`payment` | 12
+`pay-statement` | 12
+
+If an application rate limit is encountered, it will contain the `finch_code`: [finch_application_rl](/docs/Development-Guides/Errors/Error-Types.md#error-types-1) in the response body.
+
+```json
+// HTTP 429 response body for application rate limit exceeded
+{
+    "statusCode": 429,
+    "status": 429,
+    "code”: 429,
+    "message": "Too many requests for token",
+    "name": "rate_limit_exceeded_error",
+    "finch_code": "finch_application_rl"
+}
+```
+
+## IP Address Rate Limits
+
+Finch also enforces individual IP Address rate limits. An IP address is a unique address that identifies a device on the internet or a local network. If the number of `max requests` are sent from the same IP Address within the `duration` time set, a `penalty` is enforced. No more requests are allowed once the penalty is enforced. Once the penalty duration is complete, requests will be accepted again.
+
+Type | Max requests | Duration | Penalty
+-------|-------------|-------|-------------
+`API` | 1000 | 5 minutes | 60 minutes
+
+If an IP Address rate limit is encountered, it will contain the `finch_code`: [finch_api_ip_rl](/docs/Development-Guides/Errors/Error-Types.md#error-types-1) in the response body.
+
+```json
+// HTTP 429 response body for application rate limit exceeded
+{
+    "statusCode": 429,
+    "status": 429,
+    "code”: 429,
+    "message": "Too many requests for token",
+    "name": "rate_limit_exceeded_error",
+    "finch_code": "finch_api_ip_rl"
+}
+```
+
+***
+
```

Comment:
do we want to call out anywhere here that in addition to these rate limits, we can also return 429s if an upstream provider rate limits us?